utilities/virt.py

```
@@ -2004,19 +2004,31 @@ def vm_instance_from_template(
+    Waits for KubeVirt conditions to stabilize after node operations to ensure
+    cluster health before proceeding with test execution or teardown.
+
+    Args:
+        node: The node to cordon/drain.
+        node_mgmt: Operation type - "cordon" or "drain".
+        admin_client: DynamicClient with cluster-admin privileges.
+                      Required for node operations and reading KubeVirt/HCO conditions.
+    """
+    hco_namespace = get_hco_namespace(admin_client=admin_client)
     try:
         LOGGER.info(f"{node_mgmt.capitalize()} the node {node.name}")
```

Ruff (0.14.13)

Logging statement uses f-string

(G004)